docs/sdks/uid2-sdk-ref-python.md

```diff
@@ -67,57 +63,149 @@ The package is published in this location:
+
+| Property      | Description                                                                                                                                     |
+|:--------------|:------------------------------------------------------------------------------------------------------------------------------------------------|
+| `status`      | The decryption result status. For a list of possible values and definitions, see [Decryption Response Statuses](#decryption-response-statuses). |
+| `uid`         | The raw UID2 for the corresponding UID2 token.                                                                                                  |
+| `established` | The timestamp indicating when a user first established the UID2 with the publisher.                                                             |
+
+### Decryption Response Statuses
+
+| Value                      | Description                                                             |
+|:---------------------------|:------------------------------------------------------------------------|
+| `SUCCESS`                  | The UID2 token was decrypted successfully and a raw UID2 was returned.  |
+| `NOT_AUTHORIZED_FOR_KEY`   | The requester does not have authorization to decrypt this UID2 token.   |
+| `NOT_INITIALIZED`          | The client library is waiting to be initialized.                        |
+| `INVALID_PAYLOAD`          | The incoming UID2 token is not a valid payload.                         |
+| `EXPIRED_TOKEN`            | The incoming UID2 token has expired.                                    |
+| `KEYS_NOT_SYNCED`          | The client has failed to synchronize keys from the UID2 service.        |
+| `VERSION_NOT_SUPPORTED`    | The client library does not support the version of the encrypted token. |
+| `DOMAIN_NAME_CHECK_FAILED` | The domain name doesn't match the the domain of the encrypted token.    |
+| `INVALID_TOKEN_LIFETIME`   | The lifetime of the incoming UID2 token is longer than expected.        |
+
+## Usage for DSPs
+
+The following instructions provide an example of how you can decode bid stream tokens using the UID2 SDK for Python as a DSP.
+
+1. Create a `BidstreamClient`:
+
+```py
+client = BidstreamClient(UID2_BASE_URL, UID2_API_KEY, UID2_SECRET_KEY)
+```
+
+2. Refresh once at startup, and then periodically (recommended refresh interval is hourly):
+
+```py
+client.refresh()
+```
+
+3. Decrypt a token into a raw UID2. Pass the domain name of the site where the bid originated from:
+
+```py
+decrypted = client.decrypt_token_into_raw_uid(uid_token, domain)
+# If decryption succeeded, use the raw UID2.
+if decrypted.success:
+    #  Use decrypted.uid
+else:
+   # Check decrypted.status for the failure reason.
+```
+
+For a full example, see the `sample_bidstream_client.py` in [examples/sample_bidstream_client.py](https://github.com/IABTechLab/uid2-client-python/blob/main/examples/sample_bidstream_client.py).
+
+## Usage for UID2 Sharers
```

Comment:
could we keep the Sharers section at the bottom like before? I'm not sure what the rationale is? But usage for Publishers definitely should be higher than Sharer for now.

Reply:
Hmm. Will move this